tt_metal/common/metal_soc_descriptor.cpp

```diff
@@ -109,10 +109,22 @@ CoreCoord metal_SocDescriptor::get_physical_tensix_core_from_logical(const CoreC
     return physical_tensix_core;
 }
 
+CoreCoord metal_SocDescriptor::get_physical_dram_core_from_logical(const CoreCoord &logical_coord) const {
+    TT_FATAL(
+        (logical_coord.x < this->get_num_dram_channels()) and
+        (logical_coord.y == 0),
+        "Bounds-Error -- Logical_core={} is outside of logical_grid_size={}",
+        logical_coord.str(),
+        CoreCoord(1, this->get_num_dram_channels())
+    );
+    return this->get_preferred_worker_core_for_dram_channel(logical_coord.x);
```

Comment:
I remember a discussion about GS vs WH dram locations being vertical vs horizontal.

Why only use x and not y, based on the architecture, and what documentation would support that for each architecture (GS, WH, BH)?

Reply:
hey, this has been merged to main, but the reason is that we think of the logical dram grid as 1d. and this helps a lot when using the current sharding infra